mindsdb/integrations/handlers/autokeras_handler/autokeras_handler.py

```diff
@@ -0,0 +1,158 @@
+import random
+import string
+from typing import Optional
+import os
+import pandas as pd
+import numpy as np
+import autokeras as ak
+from sklearn import preprocessing
+from mindsdb.integrations.libs.base import BaseMLEngine
+from tensorflow.keras.models import load_model
+
+
+# Makes this run on Windows Subsystem for Linux
+os.environ["XLA_FLAGS"] = "--xla_gpu_cuda_data_dir=/usr/lib/cuda"
+
+trainer_dict = {"regression": ak.StructuredDataRegressor, "classification": ak.StructuredDataClassifier}
+
+DEFAULT_TRIALS = 100
+
+
+def train_model(df, target, max_trials=DEFAULT_TRIALS):
+    """Helper function to trains an AutoKeras model with an input df.
+
+    Automatically decides on classification vs. regression depending on
+    the type of the target.
+
+    Will auto-encode categorical variables as dummies.
+
+    Returns both trained model and the names of categoric dummy columns, which
+    are passed later to the prediction method.
+    """
+    # Choose regressor of classifier based on target data type
+    if np.issubdtype(df[target].dtype, np.number):
+        mode = "regression"
+        y_train = df[target]
+    else:
+        mode = "classification"
+        lb = preprocessing.LabelBinarizer()
+        y_train = lb.fit_transform(df[target])
+
+    training_df = df.drop(target, axis=1)
+    trainer = trainer_dict[mode](overwrite=True, max_trials=max_trials)
+
+    # Save the column names of all numeric columns before transforming any categorical columns into dummies
+    numeric_column_names = training_df.select_dtypes(include=[np.number]).columns.values.tolist()
+    training_df = pd.get_dummies(training_df)
+    categorical_dummy_column_names = [
+        col for col in training_df.columns.values.tolist() if col not in numeric_column_names
+    ]
+    trainer.fit(training_df, y_train, verbose=2)
+    return trainer.export_model(), categorical_dummy_column_names
+
+
+def get_prediction_df(mindsdb_df, training_df):
```

Comment:
I may be misreading this method, but wouldn't adding back data from missing columns from `training_df` into `mindsdb_df` be potentially catastrophic for the prediction (much like missing data as we discussed previously)? As in, mismatched rows would mean a fairly arbitrary imputation when the column is missing.

Unless the above is completely misinterpreting what this does, I would recommend removing this method, and instead simply requiring that all columns are being passed by the user (both with `WHERE` and `JOIN`, which the handler can't really differentiate either) inside `predict()`, otherwise just raise an exception. This has the added benefit of avoiding storage of `training_df`, which is potentially a large file.

Reply:
This is indeed related to the missing data issue. We wanted users to be able to make predictions without specifying all columns in the `WHERE` clause, so we would fill in columns they didn't specify with the best match from the training data. I don't think it would be catastrophic, because the imputation would only happen if we could find a match in the training data, but it's not very elegant.

Happy to remove this and just make users specify all columns when making predictions as you suggest. This was probably some over-engineering on our end ^^

Reply:
Yep, I think probably best to opt for the simplest route here to avoid unintuitive behavior. If users repeatedly request a way to deal with missing data, we can bring this back and improve it, or perhaps the imputation system in `dataprep_ml`.

Reply:
Ye now that I think about this, the batch prediction option should take care of the user behaviour in this case. 

This should be fixed in the latest commit.